merge.go

```diff
@@ -151,13 +163,27 @@ func (fs *mergableFiles) findOutfile(f *File) *File {
 				return fs.outfiles[i]
 			}
 		}
-		// Record a newly mergable file we can use in future merge attempts
-		fs.outfiles = append(fs.outfiles, f)
-		return f
+		// Record a newly mergable File/FileHeader we can use in future merge attempts
+		outf := NewFile()
+		outf.Header = f.Header
+		outf.Control = f.Control
+		fs.outfiles = append(fs.outfiles, outf)
+		return outf
 	}
 	return lookup(0)
 }
 
+func mergeEntries(b1, b2 Batcher) Batcher {
+	b, _ := NewBatch(b1.GetHeader())
+	entries := sortEntriesByTraceNumber(append(b1.GetEntries(), b2.GetEntries()...))
+	for i := range entries {
+		b.AddEntry(entries[i])
+	}
+	b.SetControl(b1.GetControl())
+	b.Create()
```

Comment:
This should never err right?

Reply:
Correct. We can check that error - probably should.